cost/google/idle_vm_recommendations/google_vm_recommendations.pt

```diff
@@ -109,10 +109,63 @@ datasource "ds_google_project" do
   end
 end
 
+script "js_google_project", type: "javascript" do
+  parameters "param_project"
+  result "request"
+  code <<-EOS
+  var filter = "(lifecycleState:ACTIVE)"
+
+  // Apply filter by project only if "param_project" was set.
+  if (param_project.length != 0) {
+    // Since we already have a filter applied this AND is required to apply new filter.
+    filter += " AND ("
+    _.each(param_project, function (id, index) {
+      filter += '(projectId = "' + id + '")'
+
+      // Adds OR in case it isn't last item, means there are more items to be added.
+      if ((index + 1) != param_project.length) {
+        filter += " OR "
+      }
+    })
+    filter += ")"
+  }
+
+  var request = {
+    auth: "auth_google",
+    pagination: "google_pagination",
+    host: "cloudresourcemanager.googleapis.com",
+    path: "/v1/projects/",
+    query_params: {
+      "filter": filter
+    }
+  }
+  EOS
+end
+
 datasource "ds_recommenders" do
```

Comment:
I would add a description to each datasource describing what it is doing. This makes it easier for someone to follow what the policy template is doing at each step. You can use the [Rightsize VM Recommender policy template](https://github.com/flexera-public/policy_templates/blob/master/cost/google/rightsize_vm_recommendations/google_rightsize_vm_recommendations.pt) as a reference, as that policy is similar to this one and shares a lot of the same datasources/scripts.

```suggestion
#CREATE LIST OF RECOMMENDER DETAILS FOR RECOMMENDER API CALL USING COMBINED VM & CPU UTILIZATION DATA
datasource "ds_recommenders" do
```

Reply:
@nia-vf1 thank you :), I added descriptions to datasources.